### PR TITLE
Penjualan container

### PR DESCRIPTION
**Pull request by Muh-Baharuddin**
@sidratul please review my code, separate the container penjualan into several small parts and create pagination

### CODE REVIEW

File: src/containers/Penjualan/Table/TablePenjualan.tsx [hunk 1]
```diff
@@ -28,10 +29,21 @@ type handleShowType = {
 
 const TablePenjualan = (props: handleShowType) => {
   const { showAdd, showEdit, setShowAdd, setShowEdit } = props
-  const [cookies] = useCookies(['token', 'user'])
+  const { user } = useMarContext()
+  const [cookies] = useCookies(['token'])
```
Comment (sidratul) [inline src/containers/Penjualan/Table/TablePenjualan.tsx:33]: why need cookie? save token also in contex. better, when login, save default header token to axios. so you dont have to send header for each request. 
Read:
https://axios-http.com/docs/config_defaults

File: src/containers/Penjualan/Table/TablePenjualan.tsx [hunk 2]
```diff
@@ -45,19 +57,19 @@ const TablePenjualan = (props: handleShowType) => {
...
-
   let token = cookies.token
   let config = {
     headers: {
       Authorization: 'Bearer ' + token,
     },
   }
 
+  useEffect(() => {
+    axios.get('http://localhost:4000/nota-penjualan', config).then((response) => {
```
Comment (sidratul) [inline src/containers/Penjualan/Table/TablePenjualan.tsx:68]: create env for this base api url `http://localhost:4000/`, dont hard code

File: src/containers/Penjualan/Table/TablePenjualan.tsx [hunk 3]
```diff
@@ -45,19 +57,19 @@ const TablePenjualan = (props: handleShowType) => {
...
+      setData(response.data)
+    })
+  }, [])
+
   const handleDelete = (id: string) => {
     axios
       .delete('http://localhost:4000/nota-penjualan/' + id, config)
```
Comment (sidratul) [inline src/containers/Penjualan/Table/TablePenjualan.tsx:75]: create env for this base api url http://localhost:4000/, dont hard code

File: src/pages/penjualan.tsx [hunk 4]
```diff
@@ -1,8 +1,21 @@
...
   return (
-    <DataPenjualan />
+    <div>
+      { user?.role == "ADMIN" ?
+        <AdminLayout>
+          <DataPenjualan />
+        </AdminLayout> :
+        <KaryawanLayout>
+          <DataPenjualan />
+        </KaryawanLayout>
```
Comment (sidratul) [inline src/pages/penjualan.tsx:16]: you can create another Layout, so you dont have to create condition for every page.

File: src/containers/Penjualan/Table/TablePenjualan.tsx (hunk 1 above)
Comment (Muh-Baharuddin) [reply src/containers/Penjualan/Table/TablePenjualan.tsx]: done create default header token to axios

File: src/containers/Penjualan/Table/TablePenjualan.tsx (hunk 2 above)
Comment (Muh-Baharuddin) [reply src/containers/Penjualan/Table/TablePenjualan.tsx]: done get api from env

File: src/containers/Penjualan/Table/TablePenjualan.tsx (hunk 3 above)
Comment (Muh-Baharuddin) [reply src/containers/Penjualan/Table/TablePenjualan.tsx]: done get api from env

**Pull request merged**